source/community/reactnative/src/components/inputs/choice-set/choice-set-input.js

```diff
@@ -156,30 +177,125 @@ export class ChoiceSetInput extends React.Component {
 		return array
 	}
 
+
+	setPickerValue(addInputItem) {
+		if (this.state.pickerInputText == undefined) {
+			this.setState({
+				pickerInputText: this.getPickerSelectedValue(this.state.selectedPickerValue,
+					addInputItem)
+			})
+			this.setState({ pickerChoicesArr: this.choices })
+			return (this.getPickerSelectedValue(this.state.selectedPickerValue,
+				addInputItem))
+		}
+		else {
+			return (this.state.pickerInputText)
+		}
+	}
 	/**
 	 * @description Renders Picker component as per the json
 	 */
 	renderPickerComponent(addInputItem) {
+		if(this.payload.choicesSource == undefined)
+		{
+			return (
+				<View style={styles.containerView}>
+					{(Platform.OS === Constants.PlatformIOS) && <TouchableOpacity
+						activeOpacity={1}
+						onPress={onPress}>
+						<View style={styles.touchView}>
+							<Text
+								style={[styles.text, this.styleConfig.defaultFontConfig]}
+							>
+								{this.getPickerSelectedValue(this.state.selectedPickerValue,
+									addInputItem)
+								}
+							</Text>
+							<Image
+								style={styles.button}
+								source={require(DropDownImage)}
+							/>
+						</View>
+					</TouchableOpacity>}
+					{((Platform.OS === Constants.PlatformIOS) ? this.state.isPickerSelected : true) &&
+						<View style={styles.pickerContainer}>
+							<Picker
+								mode={'dropdown'}
+								selectedValue={this.getPickerInitialValue(addInputItem)}
+								onValueChange={
+									(itemValue) => {
+										this.setState({
+											selectedPickerValue: itemValue,
+											isError: false
+										})
+										addInputItem(this.id, { value: itemValue, errorState: false });
+									}}>
+								{this.choices.map((item, key) => (
+									<Picker.Item
+										label={item.title}
+										value={item.value} key={key}
+									/>)
+								)}
+							</Picker>
+						</View>
+					}
+				</View>
+			)
+		}
+		else
 		return (
 			<View style={styles.containerView}>
-				{(Platform.OS === Constants.PlatformIOS) && <TouchableOpacity
-					activeOpacity={1}
-					onPress={onPress}>
-					<View style={styles.touchView}>
-						<Text
-							style={[styles.text, this.styleConfig.defaultFontConfig]}
+				{(Platform.OS === Constants.PlatformIOS) && <View>
+					<View style={{ flexDirection: "row", display: "flex", width: "100%", backgroundColor: 'white', borderWidth: 1, borderColor: "black",position:'relative' }}>
+						<TextInput
+							value={this.setPickerValue(addInputItem)}
+							onChangeText={(text) => {
+								if (!this.state.isPickerSelected) {
+
+									this.setState({ isPickerSelected: true })
+								}
+								this.setState({
+									selectedPickerValue: text,
```

Comment:
Multiple setStaste inside a function is not recommended. I think we can handle this in a better way.